nannyml_cloud_sdk/monitoring/custom_metric.py

```diff
@@ -0,0 +1,218 @@
+import datetime
+from typing import TypedDict, Optional, List, Literal, overload
+
+from gql import gql
+
+from nannyml_cloud_sdk.client import execute
+from nannyml_cloud_sdk.enums import ProblemType
+
+
+class CustomMetricSummary(TypedDict):
+    """Summary of a custom metric.
+
+    Attributes:
+        id: Unique identifier of the custom metric (generated by NannyML Cloud when a custom metric is created).
+        name: User-defined name of the custom metric.
+        problemType: Type of problem the custom metric can be applied to.
+        createdAt: Timestamp when the model was created.
+    """
+    id: int
+    name: str
+    problemType: ProblemType
+    createdAt: datetime.datetime
+
+
+class CustomMetricDetails(TypedDict):
+    """Details of a custom metric.
+
+    Attributes:
+        description: User-defined description of the custom metric
```

Comment:
Guessing you didn't add the function attributes here because they may be missing depending on the metric type?
It does seem useful to add for the API docs. Maybe we could solve it using inheritance?

I think something like this could work:
``` python
class CustomMetricDetails(TypedDict):
    """Details of a custom metric.

    Attributes:
        description: User-defined description of the custom metric

    """
    description: str


class CustomClassificationMetricDetails(CustomMetricDetails):
    """Details of a custom classification metric.

    Attributes:
        calculateFn: Function to calculate the metric.
        estimateFn: Optional function to estimate the metric.
    """
    calculateFn: str
    estimateFn: Optional[str]


class CustomRegressionMetricDetails(CustomMetricDetails):
    """Details of a custom regression metric.

    Attributes:
        lossFn: Function to calculate the loss.
        aggregateFn: Function to aggregate the loss.
    """
    lossFn: str
    aggregateFn: str
```

Then in places that currently return `CustomMetricDetails`, return `Union[CustomClassificationMetricDetails, CustomRegressionMetricDetails]` instead

Reply:
Ah yes, fair point. Just continued from my WIP version, back when everything in the API was still fluid. Just forgot to add those attributes. Definitely useful to have, so will add them.